File: app/src/test/java/com/star_zero/example/androidmvvm/di/MockDefaultTaskRepository.java
package com.star_zero.example.androidmvvm.di;

import com.star_zero.example.androidmvvm.domain.task.Task;
import com.star_zero.example.androidmvvm.domain.task.TaskId;
import com.star_zero.example.androidmvvm.domain.task.TaskRepository;

import java.util.List;

import rx.Observable;

public abstract class MockDefaultTaskRepository implements TaskRepository {

    @Override
    public TaskId generateTaskId() {
        return null;
    }

    @Override
    public Observable<Boolean> save(Task task) {
        return null;
    }

    @Override
    public Observable<Boolean> update(Task task) {
        return null;
    }

    @Override
    public Observable<Boolean> delete(Task task) {
        return null;
    }

    @Override
    public Observable<List<Task>> getTasks() {
        return null;
    }
}

File: app/src/test/kotlin/com/star_zero/example/androidmvvm/di/MockDefaultTaskRepository.kt
package com.star_zero.example.androidmvvm.di

import com.star_zero.example.androidmvvm.domain.task.Task
import com.star_zero.example.androidmvvm.domain.task.TaskId
import com.star_zero.example.androidmvvm.domain.task.TaskRepository
import io.reactivex.Observable

abstract class MockDefaultTaskRepository : TaskRepository {

    override fun generateTaskId(): TaskId {
        return TaskId("")
    }

    override fun save(task: Task): Observable<Boolean> {
        return Observable.just(false)
    }

    override fun update(task: Task): Observable<Boolean> {
        return Observable.just(false)
    }

    override fun delete(task: Task): Observable<Boolean> {
        return Observable.just(false)
    }

    override fun fetchTasks(): Observable<List<Task>> {
        return Observable.just(listOf())
    }
}
